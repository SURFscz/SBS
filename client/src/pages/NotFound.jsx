import React from "react";
import I18n from "i18n-js";
import "./NotFound.scss";
import {isEmpty, pseudoGuid, stopEvent} from "../utils/Utils";
import {getParameterByName} from "../utils/QueryParameters";
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {Planet} from 'react-kawaii'

export default function NotFound({currentUser}) {
    return (
        <div className="mod-not-found">
            <div className="title">
                <p dangerouslySetInnerHTML={{__html: I18n.t("not_found.description_html")}}/>
                {currentUser.guest && <a href="/login" onClick={e => {
                    stopEvent(e);
                    const state = getParameterByName("state", window.location.search);
                    const guid = pseudoGuid();
                    const queryParameter = isEmpty(state) ? `?guid=${guid}` : `?guid=${guid}&state=${encodeURIComponent(state)}`;
                    window.location.href = `/login${queryParameter}`;
                }}><FontAwesomeIcon icon="arrow-right"/>{I18n.t("not_found.loginLink")}</a>}
            </div>
            <div className="content">
                <Planet size={80} mood="sad" color="white"/>
                <p className="hero">{I18n.t("not_found.title")}</p>
            </div>
        </div>
    );

}
