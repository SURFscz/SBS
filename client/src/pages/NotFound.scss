@import "../stylesheets/vars.scss";

.mod-not-found {

    background-color: $background;
    margin-bottom: $medium-padding;

    div.title {
        font-size: 22px;
        color: $dark-grey;
        max-width: $medium;
        margin: 20px auto;
        padding: 10px 0 5px 0;

        p {
            padding: 10px 0 10px 0;
        }

        a {
            display: inline-block;
            font-size: 18px;
            text-transform: uppercase;
            text-decoration: none;
            margin: 10px 0;
            color: $blue;

            &:hover {
                font-weight: bolder;
            }

            svg {
                margin-right: 10px;
            }
        }
    }

    .content {
        max-width: $medium;
        box-shadow: 0 2px 4px 0 rgba(0, 0, 0, .1);
        margin: 0 auto 25px auto;
        background-color: white;
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
        padding: 80px;

        p.hero {
            font-size: 88px;
            margin: 80px 0;
        }
    }
}
