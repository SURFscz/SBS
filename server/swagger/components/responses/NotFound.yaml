---
name: The specified resource was not found
content:
  application/json:
    schema:
      import: './swagger/schemas/Error.yaml'
    example:
      code: 404
      message: "Not found"
