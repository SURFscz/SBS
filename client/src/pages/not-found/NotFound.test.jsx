import React from "react";
import {render} from "@testing-library/react";
import NotFound from "./NotFound";

const mockConfig = {baseUrl: "http://example.com"};

describe("NotFound", () => {
    it("matches snapshot with default props", () => {
        const {asFragment} = render(
            <NotFound
                config={mockConfig}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("matches snapshot with custom title", () => {
        const {asFragment} = render(
            <NotFound
                title="Custom Title"
                config={mockConfig}
            />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("matches snapshot with custom description", () => {
        const {asFragment} = render(
            <NotFound
                description="Custom Description"
                config={mockConfig}
            />);
        expect(asFragment()).toMatchSnapshot();
    });
});
