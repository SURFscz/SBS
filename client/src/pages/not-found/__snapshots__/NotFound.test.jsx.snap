// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound matches snapshot with custom description 1`] = `
<DocumentFragment>
  <div
    class="mod-not-found"
  >
    <div
      class="mod-inner-not-found"
    >
      <p
        class="not-found-msg"
      >
        We can't seem to find that page. Sorry for that.
        <br />
        <br />
        Start over on the 
        <a
          href="[missing \\"{{base_url}}\\" value]"
        >
          homepage
        </a>
         or contact us at 
        <a
          href="mailto:sram-support@surf.nl"
        >
          sram-support@surf.nl
        </a>
        .
      </p>
      <svg>
        undraw_page_not_found.svg
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NotFound matches snapshot with custom title 1`] = `
<DocumentFragment>
  <div
    class="mod-not-found"
  >
    <div
      class="mod-inner-not-found"
    >
      <p
        class="not-found-msg"
      >
        We can't seem to find that page. Sorry for that.
        <br />
        <br />
        Start over on the 
        <a
          href="[missing \\"{{base_url}}\\" value]"
        >
          homepage
        </a>
         or contact us at 
        <a
          href="mailto:sram-support@surf.nl"
        >
          sram-support@surf.nl
        </a>
        .
      </p>
      <svg>
        undraw_page_not_found.svg
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NotFound matches snapshot with default props 1`] = `
<DocumentFragment>
  <div
    class="mod-not-found"
  >
    <div
      class="mod-inner-not-found"
    >
      <p
        class="not-found-msg"
      >
        We can't seem to find that page. Sorry for that.
        <br />
        <br />
        Start over on the 
        <a
          href="[missing \\"{{base_url}}\\" value]"
        >
          homepage
        </a>
         or contact us at 
        <a
          href="mailto:sram-support@surf.nl"
        >
          sram-support@surf.nl
        </a>
        .
      </p>
      <svg>
        undraw_page_not_found.svg
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
